lib/src/androidTest/java/tech/relaycorp/relaydroid/persistence/EncryptedDiskPersistenceTest.kt

```diff
@@ -0,0 +1,79 @@
+package tech.relaycorp.relaydroid.persistence
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.nio.charset.Charset
+
+@RunWith(AndroidJUnit4::class)
+class EncryptedDiskPersistenceTest {
+
+    private val coroutineScope = TestCoroutineScope()
+    private val subject =
+        EncryptedDiskPersistence(
+            ApplicationProvider.getApplicationContext(),
+            coroutineScope.coroutineContext
+        )
+
+    @After
+    fun tearDown() {
+        coroutineScope.runBlockingTest {
+            subject.deleteAll()
+        }
+    }
+
+    @Test
+    fun getAndSet() = coroutineScope.runBlockingTest {
```

Comment:
Can you please nest tests for `get` and add a test for when the file doesn't exist?

Reply:
Ditto for `set` -- it's important to observe what happens when the file exists already and when it doesn't. But most importantly, please write a test to ensure we're actually encrypting things as expected.

Reply:
This is an instrumented test, we don't have nesting and pretty display names.

Reply:
Good point -- sorry, I missed the fact it was under `androidTest`.

But now I know that, I think this should be moved to the unit test folder:

- I think that's what it is with the latest changes, given the thoroughness in the cases it's covering.
- I think instrumentation tests are more useful in checking high-level tasks using the public API -- Not internal classes doing low-level operations, like this one.

Whether we call it "unit" or "instrumentation" is usually irrelevant, but I think it matters in this library because we have a target coverage with unit tests and no target coverage with instrumentation tests -- And that raises a separate point: How come JaCoCo didn't complain about the lack of unit test coverage for this class?

Reply:
We're testing this together with the EncryptedFile and MasterKey components of Android. Those can't be mocked, Robolectric does not support them at the moment. Is you do prefer this to be a unit test, I need to abstract those away, and we would only be testing the interaction with them. But I think this would become a much weaker test.

We've disabled the jacoco verification on the last PR, until we finish all components.

Reply:
Doesn't Roboelectric work in unit tests?

> We've disabled the jacoco verification on the last PR, until we finish all components.

True! My bad.